src/test/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoServiceSpec.scala

```diff
@@ -313,4 +260,16 @@ class LeonardoServiceSpec extends TestKit(ActorSystem("leonardotest")) with Flat
     leo.listClusters(Map("_labels" -> "bogus")).failed.futureValue shouldBe a [ParseLabelsException]
     leo.listClusters(Map("_labels" -> "a,b")).failed.futureValue shouldBe a [ParseLabelsException]
   }
+
+  it should "delete the init bucket if cluster creation fails" in isolatedDbTest {
+    // create the cluster
+    val clusterCreateResponse = leo.createCluster(googleProject, gdDAO.badClusterName, testClusterRequest).failed.futureValue
+
+    clusterCreateResponse shouldBe a [CallToGoogleApiFailedException]
+
+    // check the firewall rule was created for the project
+    gdDAO.firewallRules should contain (googleProject, dataprocConfig.clusterFirewallRuleName)
```

Comment:
Curious, why are we checking that the firewall rule was created?

Reply:
I copied it from other tests. :) Even if cluster creation fails it'll still create the firewall rule, which I think is harmless.